LabSources/CustomRoles/CM-1902/Invoke-UpdateCM.ps1

```diff
@@ -0,0 +1,518 @@
+                Write-ScreenInfo -Message "."
+                Write-ScreenInfo -Message "Download did not start, restarting SMS_EXECUTIVE" -TaskStart -Type "Warning"
+                try {
+                    Restart-ServiceResilient -ComputerName $SccmServerName -ServiceName "SMS_EXECUTIVE" -ErrorAction "Stop" -ErrorVariable "RestartServiceResilientErr"
+                }
+                catch {
+                    $Message = "Could not restart SMS_EXECUTIVE ({0})" -f $RestartServiceResilientErr.ErrorRecord.Exception.Message
+                    Write-ScreenInfo -Message $Message -TaskEnd -Type "Error"
+                    throw $Message
+                }
+                Write-ScreenInfo -Message "Activity done" -TaskEnd
+            } -ScriptBlock {
+                $job = Invoke-LabCommand -ActivityName "Verifying update download initiated OK" -Variable (Get-Variable -Name "Update", "SccmSiteCode") -ScriptBlock {
+                    $Query = "SELECT * FROM SMS_CM_UPDATEPACKAGES WHERE PACKAGEGUID = '{0}'" -f $Update.PackageGuid
+                    Get-CimInstance -Namespace "ROOT/SMS/site_$SccmSiteCode" -Query $Query -ErrorAction "Stop"
+                }
+                Wait-LWLabJob -Job $job -NoNewLine
+                try {
+                    $Update = $job | Receive-Job -ErrorAction "Stop" -ErrorVariable "ReceiveJobErr"
+                }
+                catch {
+                    Write-ScreenInfo -Message ("Failed to query SMS_CM_UpdatePackages after initiating download (2) ({0})" -f $ReceiveJobErr.ErrorRecord.Exception.Message) -TaskEnd -Type "Error"
+                    throw $ReceiveJobErr
+                }
+            }
+        }
+        # Writing dot because of -NoNewLine in Wait-LWLabJob
+        Write-ScreenInfo -Message "."
+        Write-ScreenInfo -Message "Download started"
+        Write-ScreenInfo -Message "Activity done" -TaskEnd
+
+    }
+    #endregion
+    
+    #region Wait for update to finish download
+    if ($Update.State -eq [SMS_CM_UpdatePackages_State]::Downloading) {
+
+        Write-ScreenInfo -Message "Waiting for update to finish downloading" -TaskStart
+        $Update = New-LoopAction -LoopTimeout 604800 -LoopTimeoutType "Seconds" -LoopDelay 15 -LoopDelayType "Seconds" -ExitCondition {
+            $Update.State -eq [SMS_CM_UpdatePackages_State]::ReadyToInstall
+        } -IfTimeoutScript {
+            # Writing dot because of -NoNewLine in Wait-LWLabJob
+            Write-ScreenInfo -Message "."
+            $Message = "Download timed out"
+            Write-ScreenInfo -Message $Message -TaskEnd -Type "Error"
+            throw $Message
+        } -IfSucceedScript {
+            # Writing dot because of -NoNewLine in Wait-LWLabJob
+            Write-ScreenInfo -Message "."
+            return $Update
+        } -ScriptBlock {
+            $job = Invoke-LabCommand -ActivityName "Querying update download status" -Variable (Get-Variable -Name "Update", "SccmSiteCode") -ScriptBlock {
```

Comment:
Did you happen to see what the status was in console for the update while it was 131075?

Reply:
The SQL native client update is not a show stopper, or at least it shouldn't be. It's just a warning since 1810 IIRC. It adds TLS 1.2 support. However, of course, I will add it, because why not. I'm just curious why you were indefinitely sat at this state. Did you happen to look at cmupdate.log?

Reply:
Ignore my previous q. re checking cmupdate.log. At this point site update isn't triggered. I'll add SQL native client anyway and see what happens. Odd I didn't experience this.

Reply:
Maybe different server builds? I've used Server 2019 Standard 10.0.17763.379 for my deployment.